### ISSUE TITLE
object-shorthand rule incorrectly marks computed object property as an error

### ISSUE THREAD

**Issue by myme** [eslint/eslint#6154]
**What version/configuration are you using?**

`package.json`
```json
{
  "name": "eslint-object-shorthand-bug",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "eslint ."
  },
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "babel-eslint": "^6.0.4",
    "eslint": "^2.9.0",
    "eslint-config-airbnb": "^9.0.1",
    "eslint-plugin-import": "^1.8.0"
  },
  "eslintConfig": {
    "extends": "airbnb/base",
    "parser": "babel-eslint"
  }
}
```

`repro.js`
```javascript
export default function addType(obj, type) {
  return Object.assign({ [type]: type }, obj);
}
```

**What did you do?**

`npm test`

**What happened?**

```bash
> eslint-object-shorthand-bug@1.0.0 test /home/mmyrseth/try-bin/js/eslint
> eslint .


~/eslint-object-shorthand-bug/repro.js
  2:26  error  Expected property shorthand  object-shorthand

✖ 1 problem (1 error, 0 warnings)

npm ERR! Test failed.  See above for more details.
```

**What did you expect to happen?**

The object shorthand is intended to simplify cases where you have object literals like the following: `{ type: type }` where you can replace it with simply: `{ type }`. However, this has a completely different meaning than what's intended with the computed/dynamic property names: `{ [type]: type }`. Here we do not want to set the key "type", but the key whose name is contained in the `type` variable (and we want to set the value to the same thing, so `key === value`).

**Comment by mysticatea** [comment]
Thank you for this issue.

This is duplicate of #6015.
It has been fixed on master branch. 
Please wait for the next release (within a few days).

**Comment by myme** [comment]
Sorry about that. I only searched through open bugs before submitting ;) Great to see that it's fixed!
